apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ActiveStack.java

```diff
@@ -80,7 +80,7 @@ public ElasticContext<?> currentContext() {
         if (current instanceof ElasticContextWrapper) {
             return ((ElasticContextWrapper<?>) current).getWrappedContext();
         }
-        return current;
+        return current != null ? current : EmptyElasticContext.INSTANCE;
```

Comment:
This change has been to simplify plugins: Instead of returning null when there is no context, we return an empty context. This avoids a lot of null-checks in the plugins.
If one really needs to know wheter a context is active or not, it is possible to do so by calling `ElasticContext.isEmpty()`. This is for example useful to optimize intra-process context propagation: No need to propagate empty contexts here.